pkg/backend/local/rows.go

```diff
@@ -354,12 +356,24 @@ func (data *resourceRowData) getInfo() string {
 		appendDiagMessage(fmt.Sprintf("%v debug messages", diagInfo.DebugCount))
 	}
 
-	// If we're not totally done, also print out the worst diagnostic next to the status message.
-	// This is helpful for long running tasks to know what's going on.  However, once done, we print
-	// the diagnostics at the bottom, so we don't need to show this.
-	worstDiag := getWorstDiagnostic(data.diagInfo)
-	if worstDiag != nil && !data.display.Done {
-		eventMsg := data.display.renderProgressDiagEvent(*worstDiag, true /*includePrefix:*/)
+	if !data.display.Done {
+		// If we're not totally done, and we're in the tree-view also print out the worst diagnostic
+		// next to the status message. This is helpful for long running tasks to know what's going
+		// on. However, once done, we print the diagnostics at the bottom, so we don't need to show
+		// this.
+		//
+		// if we're not in the tree-view (i.e. non-interactive mode), then we want to print out
+		// whatever the last diagnostics was that we got.  This way, as we're hearing about
+		// diagnostic events, we're always printing out the last one.
+
+		var diagnostic *engine.DiagEventPayload
+		if data.display.isTerminal {
+			diagnostic = data.diagInfo.LastDiag
+		} else {
+			diagnostic = getWorstDiagnostic(data.diagInfo)
+		}
+
+		eventMsg := data.display.renderProgressDiagEvent(*diagnostic, true /*includePrefix:*/)
```

Comment:
It's not obvious to me here how we know `diagnostic` is not `nil` at this point, and indeed the old code seemed to check for that.

Does this actually work if this code is ever called before any diagnostics has been issued?

Reply:
nope, that's a good catch.  i screwed that up.